vdirsyncer/storage/dav.py

```diff
@@ -587,12 +587,30 @@ async def _put(self, href, item, etag):
     async def update(self, href, item, etag):
         if etag is None:
             raise ValueError("etag must be given and must not be None.")
-        href, etag = await self._put(self._normalize_href(href), item, etag)
+        try:
+            href, etag = await self._put(self._normalize_href(href), item, etag)
+        except aiohttp.ClientResponseError as e:
+            if e.status == 409:
+                dav_logger.debug("Conflict, will delete old event and recreate it.")
+                await self.delete(self._normalize_href(href), None)
+                dav_logger.debug("Now trying again")
+                href, etag = await self._put(self._normalize_href(href), item, None)
+            else:
+                raise e
```

Comment:
In my case, I have the following `~/.config/vdirsyncer/config`:

```
[general]
status_path = "~/.cache/vdirsyncer/status/"

[storage caldav]
type = "caldav"
url = # redacted
username = # redacted
password = # redacted
read_only = true
item_types = ["VEVENT"]

[storage gcal]
type = "google_calendar"
token_file = "~/.cache/vdirsyncer/gcal_robert_work.token"
client_id = # redacted
client_secret = # redacted
item_types = ["VEVENT"]

[pair caldav_to_gcal]
a = "caldav"
b = "gcal"

# Robert work (new)
collections = [["caldav_to_gcal", "a_calendar_id", "b_calendar_id@group.calendar.google.com"]]

conflict_resolution = "a wins"
partial_sync = "ignore"
```

So in my case, since my caldav server is set to read-only and I have conflict resolution set to `a wins`, I actually don't care about the old event, and I'm completely fine with deleting it.